pennylane/ops/functions/map_wires.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the qml.map_wires function.
+"""
+from functools import wraps
+from typing import Callable, Union
+
+import pennylane as qml
+from pennylane.measurements import MeasurementProcess
+from pennylane.operation import Operator
+from pennylane.qnode import QNode
+from pennylane.queuing import QueuingManager
+from pennylane.tape import QuantumTape
+
+
+def map_wires(
+    input: Union[Operator, MeasurementProcess, QuantumTape, QNode, Callable],
+    wire_map: dict,
+    queue=False,
+    replace=False,
+):
+    """Changes the wires of an operator, tape, qnode or quantum function according to the given
+    wire map.
+
+    Args:
+        input (.Operator, pennylane.QNode, .QuantumTape, or Callable): an operator, quantum node,
+            tape or function that applies quantum operations
+        wire_map (dict): dictionary containing the old wires as keys and the new wires as values
+        queue (bool): Whether or not to queue the object when recording. Defaults to False.
+        replace (bool): When ``queue=True``, if ``replace=True`` the input operators will be
+            replaced by its mapped version. Defaults to False.
+
+    Returns:
+        (.Operator, pennylane.QNode, .QuantumTape, or Callable): input with changed wires
+
+    **Example**
+
```

Comment:
Since we aren't supporting decorator syntax for the time being, we should add a note or warning that `@qml.map_wires` *won't* work.

Reply:
@albi3ro we could always use the approach JAX uses in their documentation. E.g.,

```python
@functools.partial(qml.map_wires, wire_map)
@qnode
def func(x):
   ...
```

But not too important I agree

Reply:
Added note mentioning the use of functools